include/seqan3/std/iterator

```diff
@@ -55,6 +55,73 @@
 namespace std
 {
 
+/*!\defgroup iterator_associated_types Associated types
+ * \ingroup iterator
+ * \brief Iterator requirements
+ */
+
+/*!\defgroup incrementable_traits Incrementable traits
+ * \ingroup iterator_associated_types
+ * \brief Traits for objects that are WeaklyIncrementable.
+ */
+
+//!\addtogroup incrementable_traits
+//!\{
+
+/*!\brief A type trait class that provides uniform interface to the properties of types that model the
+ *        WeaklyIncrementable concept.
+ * \tparam t The type to query.
+ * \sa https://en.cppreference.com/w/cpp/iterator/incrementable_traits
+ */
+template <typename t>
+struct incrementable_traits
+{};
+
+//!\cond
+template <typename t>
+    requires std::is_object_v<t>
+struct incrementable_traits<t*>
+{
+    using difference_type = ptrdiff_t;
+};
+
+template <typename t>
+struct incrementable_traits<const t> : incrementable_traits<t>
+{};
+
+template <typename t>
+    requires requires { typename t::difference_type; }
+struct incrementable_traits<t>
+{
+    using difference_type = typename t::difference_type;
+};
+
+template <typename t>
+    requires (!requires { typename t::difference_type; } &&
+              requires { typename iterator_traits<t>::difference_type; })
```

Comment:
1) Why this specialisation? I don't see it at cppreference (https://en.cppreference.com/w/cpp/iterator/incrementable_traits). 

2) Is this maybe ambiguous to the specialisation below? (If a type has `typename iterator_traits<t>::difference_type` but also fulfills `requires(t const & a, t const & b) { requires Integral<decltype(a - b)>; }`) ?